docs/src/tutorials/Conic programs/logistic_regression.jl

```diff
@@ -0,0 +1,235 @@
+# the equivalent formulation
+#
+# ```math
+# \left\{
+# \begin{aligned}
+# (u_i -t_i , 1, z_{i1}) & \in  K_{exp}  \\
+# (-t_i , 1, z_{i2}) & \in  K_{exp}  \\
+# z_{i1} + z_{i2} & \leq  1
+# \end{aligned}
+# \right.
+# ```
+#
+# In this setting, the conic version of the logistic regression problems writes out
+#
+# ```math
+# \begin{aligned}
+# \min_{t, z, r, \theta}&  \; \sum_{i=1}^n t_i + \lambda r \\
+# \text{subject to } & \quad  (u_i -t_i , 1, z_{i1})  \in  K_{exp}  \\
+#                    & \quad  (-t_i , 1, z_{i2})  \in  K_{exp}  \\
+#                    & \quad  z_{i1} + z_{i2}  \leq  1 \\
+#                    & \quad u_i = -y_i x_i^\top \theta \\
+#                    & \quad r \geq \|\theta\|
+# \end{aligned}
+# ```
+#
+# and thus encompasses $3n + p + 1$ variables and $3n + 1$ constraints ($u_i = -y_i \theta^\top x_i$
+# is only a virtual constraint used to clarify the notation).
+# Thus, if $n \gg 1$, we get a large number of variables and constraints.
+
+
+# ## Fitting logistic regression with a conic solver
+#
+# It is now time to pass to the implementation. We choose SCS as a conic solver.
+using JuMP
+using Random
+using SCS
+
+Random.seed!(2713);
+
+# We start by implementing a function to generate a fake dataset, and where
+# we could tune the correlation between the feature variables. The function
+# is a direct transcription of the one used in [this blog post](http://fa.bianp.net/blog/2013/numerical-optimizers-for-logistic-regression/).
+function generate_dataset(n_samples=100, n_features=10; shift=0.0)
+    X = randn(n_samples, n_features)
+    w = randn(n_features)
+    y = sign.(X * w)
+    X .+= 0.8 * randn(n_samples, n_features) # add noise
+    X .+= shift # shift the points in the feature space
+    X = hcat(X, ones(n_samples, 1))
+    return X, y
+end
+
+# We write a `softplus` function to formulate each constraint
+# $t \geq \log(1 + \exp(u))$ with two exponential cones.
+function softplus(model, t, u)
+    z = @variable(model, [1:2], lower_bound=0.0)
+    @constraint(model, sum(z) <= 1.0)
+    @constraint(model, vec([u - t, 1, z[1]]) in MOI.ExponentialCone())
+    @constraint(model, vec([-t, 1, z[2]]) in MOI.ExponentialCone())
+end
+
+# ## $\ell_2$ regularized logistic regression
+# Then, with the help of the `softplus` function, we could write our
+# optimization model. In the $\ell_2$ regularization case, the constraint
+# $r \geq \|\theta\|_2$ rewrites as a second order cone constraint.
+function build_logit_model(X, y, λ)
+    n, p = size(X)
+    model = Model()
+    @variable(model, θ[1:p])
+    @variable(model, t[1:n])
+    for i in 1:n
+        u = - (X[i, :]' * θ) * y[i]
+        softplus(model, t[i], u)
+    end
+    ## Add ℓ2 regularization
+    @variable(model, 0.0 <= reg)
+    @constraint(model, vec([reg; θ]) in MOI.SecondOrderCone(p+1))
+    ## Define objective
+    @objective(model, Min, sum(t) + λ * reg)
+    return model
+end
+
+## Generate one dataset.
```

Comment:
Prefer things like this as `# Generate ...` so they get rendered as plain-text, not code comments.